### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by orangemug**
Updated most of the dependencies, with the exception of

```
$ npm outdated
Package                    Current  Wanted  Latest  Location     Comment
babel-loader                 7.1.1   7.1.1   7.1.2  maputnik     Locked to 7.1.1 see <https://github.com/babel/babel-loader/issues/505>
jsonlint                     1.6.2     git     git  maputnik     Not required
maputnik-design              0.1.0     git     git  maputnik     Not required
ol-mapbox-style              1.0.2   1.0.2   2.6.6  maputnik     Issues that need resolving with ES6 syntax in the module
stylelint                   7.13.0  7.13.0   8.2.0  maputnik     We need to switch to a scss aware linter
stylelint-config-standard   15.0.1  15.0.1  17.0.0  maputnik     We need to switch to a scss aware linter
```

### CODE REVIEW

**Comment by orangemug** [comment]
I've fixed the failing tests and also removed nodejs v4 for the time being. As it was causing all the tests to fail. There is some odd error with the install in v4 which I'll look into later in the week.

> Removed nodejs v4 tests temporarily

**Comment by orangemug** [comment]
The latest branch build is here <https://44-84182601-gh.circle-artifacts.com/0/tmp/circle-artifacts.NgEXZgj/build/index.html>

@keum you kindly offered to test maputnik over in ticket #181. In this pull request I've updated most of the library dependencies maputnik uses. Given our lack of test coverage it'd be great if you could give it a test before I merge the changes. Basically we're looking for anything that is broken now and wasn't in the current live version at <https://maputnik.github.io/>

So basically test the app here <https://44-84182601-gh.circle-artifacts.com/0/tmp/circle-artifacts.NgEXZgj/build/index.html> and leave any feedback in this ticket. Any data will be lost so make sure you export any styles you want to keep.

**Comment by keum** [comment]
@orangemug One thing I noticed was that when Mapzen basema, Tilezen option doesn't show on yours compared to https://maputnik.github.io/editor/#12.24/47.3718/8.5481
also both in yours and original maptunik editor (maptunik.github.io/editor) when trying to add in Gallery Styles, Mapbox's basemaps (satellite, bright, basic) won't show up. I'm not sure what i'm doing wrong but I tried to load using Chrome and Firefox quantum giving same no show. 

I've added my own vector tiles from DigitalOcean and styled them and export to preview and no problem with it.

**Comment by orangemug** [comment]
> One thing I noticed was that when Mapzen basema, Tilezen option doesn't show

Good spot, that's correct, we removed that in #172 because it was no longer functional. We have a ticket to add it back in #153.

> Mapbox's basemaps (satellite, bright, basic) won't show up

That'll be due to permisisons of the mapbox key. So not an issue but thanks for raising it.

> I've added my own vector tiles from DigitalOcean and styled them and export to preview and no problem with it.

Thats what I like to hear :)

Thanks for testing @keum, the hard work is much appreciated.

**Comment by orangemug** [comment]
For future reference we also removed node.js v4/v5 support, node v5 is no longer supported and v4 was causing issues. Since v4 is in maintenance due to end April 2018 (see <https://github.com/nodejs/Release>) I decided to remove support.

There were also some breaking API changes to `@mapbox/mapbox-gl-style-spec` in a minor version release. Basically `styleSpec.latest.source_tile` was renamed. Due to travis caching of `node_modules` this didn't initially show up as a broken build. However a fresh install of maputnik failed to build (see <https://github.com/maputnik/editor/pull/192/commits/cb2f854dd5bdafdf393bf3753e6b4a8960fccf9c#diff-b901d42e1ea5b8ba1ea7465908999e83L15>)
